src/server/api/controllers/vessel-reports.controller.js

```diff
@@ -3,13 +3,50 @@
 const HttpError = require("../lib/utils/http-error");
 const knex = require("../../config/db");
 
-//Get vessels reports by vessel id
-const getVesselsReportByVoyageId = async voyage_id => {
-  const vesselReports = await knex.from("vessel_reports").where({ voyage_id });
-  if (vesselReports.length === 0) {
-    return `No voyage exist with that id ${voyage_id}.`;
+//Get vessels reports by voyage_id with limit, offset=0 and order =desc
+const getVesselsReportByVoyageIdWithParams = async (voyage_id, query) => {
+  const voyage = await knex.from("voyages").where({ id: voyage_id });
+
+  if (voyage.length === 0) {
+    throw new HttpError(
+      "Bad request",
+      `No voyage exist with that id ${voyage_id}.`,
+      404
+    );
+  }
+  if (query.orderBy === undefined) {
+    const vesselReports = await knex
+      .from("vessel_reports")
+      .where({ voyage_id })
+      .orderBy("id", "desc")
+      .limit(query.limit)
+      .offset(0);
+
+    if (vesselReports.length === 0) {
+      throw new HttpError(
+        "Bad request",
+        `No vessel-reports exist with that voyage id ${voyage_id}.`,
+        404
+      );
+    }
+    return vesselReports;
+  } else {
+    const vesselReports = await knex
+      .from("vessel_reports")
+      .where({ voyage_id })
+      .orderBy("id", query.orderBy)
+      .limit(50)
+      .offset(0);
+
+    if (vesselReports.length === 0) {
+      throw new HttpError(
+        "Bad request",
+        `No vessel-reports exist with that voyage id ${voyage_id}.`,
+        404
+      );
+    }
+    return vesselReports;
```

Comment:
Your logic does not allow me to any of the parameters together. If I do this, offset will be ignored:
```
http://localhost:3000/voyages/3/vessel-reports?limit=5&offset=2
```

If you first set default values for all of the possible query parameters, then do a single knex query with all the parameters set and without all of the if/else logic it should work.

Reply:
Also makes it a lot simpler to implement on other endpoints later.

Reply:
Working good. Thanks Zaki for recoding.